runelite-client/src/test/java/net/runelite/client/OkHttpCacheSanityTest.java

```diff
@@ -0,0 +1,125 @@
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
+import java.util.concurrent.TimeUnit;
+import net.runelite.http.api.RuneLiteAPI;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class OkHttpCacheSanityTest
+{
+	@Rule
+	public TemporaryFolder cacheFolder = new TemporaryFolder();
+
+	private static final DateTimeFormatter TIME_FMT = DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss z", Locale.US)
+		.withZone(ZoneId.of("GMT"));
+
+	private static final String BODY_404 = "<html>404 not found";
+	private static final String BODY_200 = "{success:true}";
+
+	/**
+	 * The specific thing we are trying to catch here is when a 404
+	 * response is served by cloudflare after the upstream has gotten
+	 * an updated document. CF serves the 404 response with a Date header,
+	 * but no ETag or Last-Modified. OkHttp then uses the Date header, which
+	 * is from after the upstream was edited, in a If-Modified-Since request,
+	 * which does hit the upstream, returning 304 Not Modified. Since there
+	 * is no body OkHttp serves the cached 404 as up-to-date. Better yet, since
+	 * there is now an ETag in the 304 response it requests subsequent updates
+	 * with the ETag, which will always 304 Not Modified, causing clients in
+	 * this state to get incorrect 404s until the ETag changes.
+	 */
+	@Test
+	public void testCacheSanity() throws IOException, InterruptedException
+	{
+		MockWebServer server = new MockWebServer();
```

Comment:
This must be shutdown. The few places we use mockwebserver in the codebase do this with a junit before/after, however, MockWebserver is a junit `ExternalResource` and so I believe you can instead do:
```java
@Rule
public MockWebServer server = new MockWebServer();
```